File: LegacySource/settlement/room/industry/refiner/ROOM_REFINER.cs
using System;
using System.Collections.Generic;
using System.IO;

namespace Settlement.Room.Industry
{
    public class RoomRefiner : RoomBlueprintIns<RefinerInstance>, IIndustryHaser
    {
        public const string Type = "REFINER";
        private readonly Job job;
        private readonly IList<Industry> indus;
        private readonly Constructor constructor;

        public RoomRefiner(RoomInitData init, string key, int index, RoomCategorySub cat) : base(index, init, key, cat)
        {
            constructor = new Constructor(init, this);
            PushBo(init.Data(), Type, true);
            indus = Industry.CreateIndustries(this, init, new RoomBoost[] { constructor.Efficiency }, Bonus());
            job = new Job(this, init.Data().I("STORAGE", 8, 500));
            new RoomExperienceBonus(this, init.Data(), Bonus());
            Employment().CountInputSet();
        }

        public override Furnisher Constructor()
        {
            return constructor;
        }

        protected override void Update(double ds)
        {
        }

        public override SFinderRoomService Service(int tx, int ty)
        {
            return null;
        }

        protected override void SaveP(FilePutter saveFile)
        {
            IndustryUtil.Save(saveFile, indus);
        }

        protected override void LoadP(FileGetter saveFile)
        {
            IndustryUtil.Load(saveFile, indus);
        }

        protected override void ClearP()
        {
            IndustryUtil.Clear(indus);
        }

        public override bool MakesDudesDirty()
        {
            return true;
        }

        public override void AppendView(IList<UIRoomModule> mm)
        {
        }

        public IList<Industry> Industries()
        {
            return indus;
        }
    }
}

File: settlement/room/industry/refiner/ROOM_REFINER.java
package settlement.room.industry.refiner;

import java.io.IOException;

import settlement.path.finders.SFinderRoomService;
import settlement.room.industry.module.INDUSTRY_HASER;
import settlement.room.industry.module.Industry;
import settlement.room.industry.module.IndustryUtil;
import settlement.room.industry.module.RoomBoost;
import settlement.room.main.BonusExperience.RoomExperienceBonus;
import settlement.room.main.RoomBlueprintIns;
import settlement.room.main.category.RoomCategorySub;
import settlement.room.main.furnisher.Furnisher;
import settlement.room.main.util.RoomInitData;
import snake2d.util.file.FileGetter;
import snake2d.util.file.FilePutter;
import snake2d.util.sets.LIST;
import snake2d.util.sets.LISTE;
import view.sett.ui.room.UIRoomModule;

public class ROOM_REFINER extends RoomBlueprintIns<RefinerInstance> implements INDUSTRY_HASER{

	public final static String type = "REFINER";
	final Job job;
	final LIST<Industry> indus;
	final Constructor constructor;
	
	public ROOM_REFINER(RoomInitData init, String key, int index, RoomCategorySub cat) throws IOException {
		super(index, init, key, cat);
		
		

		
		constructor = new Constructor(init, this);
		pushBo(init.data(), type, true);
		indus = Industry.createIndustries(this, init, new RoomBoost[] {constructor.efficiency}, bonus());
		job = new Job(this, init.data().i("STORAGE", 8, 500));
		new RoomExperienceBonus(this, init.data(), bonus());
		employment().countInputSet();
	}
	
	@Override
	public Furnisher constructor() {
		return constructor;
	}

	
	
	@Override
	protected void update(double ds) {
		
	}
	
	@Override
	public SFinderRoomService service(int tx, int ty) {
		// TODO Auto-generated method stub
		return null;
	}

	@Override
	protected void saveP(FilePutter saveFile){
		IndustryUtil.save(saveFile, indus);
	}
	
	@Override
	protected void loadP(FileGetter saveFile) throws IOException{
		IndustryUtil.load(saveFile, indus);
	}
	
	@Override
	protected void clearP() {
		IndustryUtil.clear(indus);
	}
	
	@Override
	public boolean makesDudesDirty() {
		return true;
	}
	
	@Override
	public void appendView(LISTE<UIRoomModule> mm) {
		
	}

	@Override
	public LIST<Industry> industries() {
		return indus;
	}

}
